src/unix/proctitle.c

```diff
@@ -50,10 +50,14 @@ char** uv_setup_args(int argc, char** argv) {
 
 #if defined(__MVS__)
   /* argv is not adjacent. So just use argv[0] */
-  process_title.str = argv[0];
-  process_title.len = strlen(argv[0]);
+  process_title.str = uv__strdup(argv[0]);
+  if (process_title.str == NULL)
+    return argv;
+  process_title.len = strlen(process_title.str);
 #else
-  process_title.str = argv[0];
+  process_title.str = uv__strdup(argv[0]);
+  if (process_title.str == NULL)
+    return argv;
```

Comment:
Can you hoist the common code (everything up to and including the if statement) out of the `#if` block?